examples/plot_gp_function.py

```diff
@@ -0,0 +1,183 @@
+from bayes_opt1 import BayesianOptimization
+from bayes_opt1 import UtilityFunction
+
+
+def normalization(utility):
+    return (utility- min(utility)) / (max(utility) - min(utility))
+
+
+def plot_gp(optimizer1, optimizer2, optimizer3, x, target, params):
+    x = x.reshape(-1,1)
+    y = target(x)
+    
+    def posterior(optimizer, x_obs, y_obs, grid):
+        optimizer._gp.fit(x_obs, y_obs)
+        mu, sigma = optimizer._gp.predict(grid, return_std=True)
+        return mu, sigma
+    
+    fig = plt.figure(figsize=(16, 10))
+    steps = len(optimizer1.space)
+    #fig.suptitle(
+        #'Gaussian Process and Utility Function After {} Steps'.format(steps),
+        #fontdict={'size':30}
+    #)
+    
+    gs = gridspec.GridSpec(2, 1, height_ratios=[2, 2]) 
+    axis = plt.subplot(gs[0])
+    acq = plt.subplot(gs[1])
+   
+    
+    x_obs1 = np.array([[res["params"]["x"]] for res in optimizer1.res])
+    y_obs1 = np.array([res["target"] for res in optimizer1.res])
+    x_obs2 = np.array([[res["params"]["x"]] for res in optimizer2.res])
+    y_obs2 = np.array([res["target"] for res in optimizer2.res])
+    x_obs3 = np.array([[res["params"]["x"]] for res in optimizer3.res])
+    y_obs3 = np.array([res["target"] for res in optimizer3.res])
+    
+    mu1, sigma1 = posterior(optimizer1, x_obs1, y_obs1, x)
+    mu2, sigma2 = posterior(optimizer2, x_obs2, y_obs2, x)
+    mu3, sigma3 = posterior(optimizer3, x_obs3, y_obs3, x)
+    axis.plot(x, y, linewidth=3, label='Target')
+    #axis.plot(x_obs.flatten(), y_obs, 'D', markersize=8, label=u'Observations', color='r')
+    axis.plot(x, mu1, '--', color='purple', label='Prediction ucb')
+    axis.plot(x, mu2, '--', color='green', label='Prediction ei')
+    axis.plot(x, mu3, '--', color='orange', label='Prediction poi')
+    
+    #axis.axvline(x=x[np.argmax(y)], linestyle=':')
+    #axis.fill(np.concatenate([x, x[::-1]]), 
+              #np.concatenate([mu1 - 1.9600 * sigma1, (mu1 + 1.9600 * sigma1)[::-1]]),
+        #alpha=.6, fc='c', ec='None',color='purple', label='95% confidence interval')
+    
+    fig.suptitle(
+        'Utility Functions After {} Steps'.format(steps-2),
+        fontdict={'size':50}
+    )
+    
+    axis.set_xlim((min(x), max(x)))
+    axis.set_ylim((None, None))
+    axis.set_ylabel('f(x)', fontdict={'size':20})
+    axis.set_xlabel('x', fontdict={'size':20})
+    
+    utility_function_ucb = UtilityFunction(kind="ucb", kappa = params['ucb']['kappa'], xi = params['ucb']['xi'])
+    utility_ucb = utility_function_ucb.utility(x, optimizer1._gp, 0)
+    
+    utility_function_ei = UtilityFunction(kind="ei", kappa = params['ucb']['kappa'], xi = params['ei']['xi'])
+    utility_ei = utility_function_ei.utility(x, optimizer2._gp, optimizer2._space.target.max()) # max(np.array([res["target"] for res in optimizer2.res])
+    
+    
+    #print(utility_ei)
+    utility_function_poi = UtilityFunction(kind="poi", kappa = params['ucb']['kappa'], xi = params['poi']['xi'])
```

Comment:
State passando `kappa = params['ucb']['kappa']` a tutte e tre le acqusizioni, non solo `ucb`

Reply:
Abbiamo messo un valore casuale perchè in quelle funzioni il parametro kappa non viene usato, ma comunque deve essere dato come input.